### PR TITLE
Revert "[SPARK-30808][SQL] Enable Java 8 time API in Thrift server"

### PR DESCRIPTION
**Pull request by yaooqinn**
This reverts commit afaeb29599593f021c9ea47e52f8c70013a4afef.

### What changes were proposed in this pull request?

Based on the result and comment from https://github.com/apache/spark/pull/27552#discussion_r385531744

In the hive module, server-side provides datetime values simply use `value.toSting`, and the client-side regenerates the results back in `HiveBaseResultSet` with `java.sql.Date(Timestamp).valueOf`. 
there will be inconsistency between client and server if we use java8 APIs

### Why are the changes needed?

the change is still unclear enough

### Does this PR introduce any user-facing change?

no
### How was this patch tested?

Nah

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan and @MaxGekk 
```
SELECT DATE_TRUNC('MILLENNIUM', DATE '1970-03-20')
```
the case passed both `SQLQueryTestSuite` and `ThriftServerQueryTestSuite` w/o that commit.

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 1]
```diff
@@ -378,6 +378,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
         localSparkSession.conf.set(SQLConf.ANSI_ENABLED.key, true)
       case _ =>
     }
+    localSparkSession.conf.set(SQLConf.DATETIME_JAVA8API_ENABLED.key, true)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: can we remove this? 

The main goal of #27552 is to remove this line. If we can remove it and still pass all tests, then #27552 can be safely removed.

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**